src/scopytitlemanager.cpp

```diff
@@ -0,0 +1,131 @@
+#include "scopytitlemanager.h"
+
+#include <QApplication>
+#include <QDebug>
+
+#define TITLE_SEPARATOR " - "
+
+using namespace scopy;
+
+ScopyTitleManager *ScopyTitleManager::pinstance_{nullptr};
+
+ScopyTitleManager::ScopyTitleManager(QObject *parent) : QObject(parent) {
+	qDebug() << "ctor";
+}
+
+ScopyTitleManager::~ScopyTitleManager() { qDebug() << "dtor"; }
+
+void ScopyTitleManager::buildTitle()
+{
+	QString result;
+	bool addSeparator = false;
+
+	if (!m_title.isEmpty()) {
+		result = m_title;
+		addSeparator = true;
+	}
+
+	if (!m_version.isEmpty()) {
+		if (addSeparator) {
+			result += TITLE_SEPARATOR;
+		}
+
+		result += m_version;
+		addSeparator = true;
+	}
+
+	if (!m_hash.isEmpty()) {
+		if (addSeparator) {
+			result += TITLE_SEPARATOR;
+		}
+
+		result += m_hash;
+		addSeparator = true;
+	}
+
+	if (!m_filename.isEmpty()) {
+		if (addSeparator) {
+			result += TITLE_SEPARATOR;
+		}
+
+		result += m_filename;
+		addSeparator = true;
+	}
+
+	m_currentTitle = result;
+	auto appWindow = qApp->activeWindow();
+	if (appWindow) {
+		appWindow->setWindowTitle(result);
+	} else {
+		qWarning() << "Cannot set Scopy title";
+	}
+}
+
+ScopyTitleManager *ScopyTitleManager::GetInstance() {
+	if (pinstance_ == nullptr) {
+		pinstance_ = new ScopyTitleManager(
+					QApplication::instance()); // singleton has the app as parent
+	} else {
+		qDebug() << "Singleton instance already created.";
+	}
+	return pinstance_;
+}
+
+void ScopyTitleManager::setName(QString title) {
```

Comment:
Consider having a static variants for these function so they can be called
ScopyTitleManager::setApplicationName(), etc